attribute-projection-functions/src/main/java/org/hypertrace/core/attribute/service/projection/functions/Equals.java

```diff
@@ -0,0 +1,17 @@
+package org.hypertrace.core.attribute.service.projection.functions;
+
+import static java.util.Objects.isNull;
+import static java.util.Objects.requireNonNullElse;
+
+import javax.annotation.Nullable;
+
+public class Equals {
+  private static final String DEFAULT_STRING = "";
+
+  public static boolean stringEquals(@Nullable String first, @Nullable String second) {
+    if (isNull(first) && isNull(second)) {
+      return true;
+    }
+    return requireNonNullElse(first, DEFAULT_STRING).equals(requireNonNullElse(second, DEFAULT_STRING));
```

Comment:
so this mans that `null === ""` - not sure that should hold. Could we do
```java
if (isNull(first) && isNull(second)) {
      return true;
 }
if (isNull(first) || isNull(second)) {
      return false;
 }
return first.equals(second);
```

Reply:
So no defaulting to empty string if null? Got it.